core/src/main/java/org/elasticsearch/index/IndexService.java

```diff
@@ -606,7 +608,11 @@ private Query parse(AliasMetaData alias, QueryShardContext shardContext) {
         try {
             byte[] filterSource = alias.filter().uncompressed();
             try (XContentParser parser = XContentFactory.xContent(filterSource).createParser(filterSource)) {
-                ParsedQuery parsedFilter = shardContext.toFilter(shardContext.newParseContext(parser).parseInnerQueryBuilder());
+                Optional<QueryBuilder> innerQueryBuilder = shardContext.newParseContext(parser).parseInnerQueryBuilder();
+                ParsedQuery parsedFilter = null;
+                if (innerQueryBuilder.isPresent()) {
+                    parsedFilter = shardContext.toFilter(innerQueryBuilder.get());
+                }
```

Comment:
Okay, so I added the early return, assuming `toFilter()` should always return a query. In case of empty inner query I think this method should still return null as `IndexService#aliasFilter()` actually documents and handles those cases and wouldn't like to break existing expectations here without further inspection.
+1 on re-visiting shard contexts `toQuery()` and `toFilter()` potentially returning null, but I'd like to do that in a separate PR.